### DISCUSSION TITLE
Policy - Hide Resource - only allow attaching / detaching

### DISCUSSION THREAD

**Discussion by jandante** [laravel/nova-issues#5832] [Q&A] [answered]
Is there a way to prevent any access (view / edit / ...) to a resource but allow for a specific user to attach them to another resource?

In short I don't want the user to see Roles in the sidebar navigation. But I do want them to be able to attach a Role to a User.

I've tried denying the user access via 'viewAny' so it doesn't appear in the sidebar. Then added the 'attachRole' to allow that user to attach roles but it's only working when 'viewAny' is true for that user.
Of am I missing something?

**Comment by treed-eaglemetal** [comment]
If you want to just hide something from the sidebar navigation, you can explicitly disable it.

https://nova.laravel.com/docs/resources/#registering-resources

> If you do not want a resource to appear in the sidebar, you may override the `displayInNavigation` property of your resource class:

```php
/**
 * Indicates if the resource should be displayed in the sidebar.
 *
 * @var bool
 */
public static $displayInNavigation = false;
```

**Reply by jandante** [reply]
Thanks for pointing that out.

However I see this is a property. In my use case depending on the role the user has it should be either true or false.
Can I make this variable?

**Reply by crynobone** [reply, marked as answer]
You should be able to use the following too:

```php
/**
 * Determine if this resource is available for navigation.
 *
 * @param  \Illuminate\Http\Request  $request
 * @return bool
 */
public static function availableForNavigation(Request $request)
{
    return $request->user()->hasRole('admin'); 
}
```

**Reply by jandante** [reply]
Thank you @crynobone
